### ISSUE TITLE
Locales support

### ISSUE THREAD

**Issue by patapizza** [facebookincubator/duckling#33]
Currently, Duckling has two different granularity levels for rules: language-wise, and common rules across languages. As an example, `AmountOfMoney` has a couple of [common rules](https://github.com/facebookincubator/duckling/blob/master/Duckling/AmountOfMoney/Rules.hs), and more specific rules for [English](https://github.com/facebookincubator/duckling/blob/master/Duckling/AmountOfMoney/EN/Rules.hs), [French](https://github.com/facebookincubator/duckling/blob/master/Duckling/AmountOfMoney/FR/Rules.hs), etc.

Today, if we want to handle country-specific forms, we'd include them under the language umbrella. Although this has worked well so far, it is not ideal. We had to include Cantonese variations for `Time` within [the Chinese rules](https://github.com/facebookincubator/duckling/commit/001ff291fc41c73ad2cbb13206586f167f7d649e) (#21).

An improvement for Duckling would be to add a finer granularity for rules: scoping by locale.
This is something we've had in mind for a while, though we are not able to prioritize it today. The goal here is to open the discussion and maybe come up with an actual implementation.

**Comment by patapizza** [comment]
On another note, there are languages not included in the two-letters ISO-639-1 standard, but have a ISO-639-3 representation (e.g. Filipino).
Some languages that have an ISO-639-1 representation are considered as [macrolanguages](https://en.wikipedia.org/wiki/ISO_639_macrolanguage) in ISO-639-3 (e.g. Chinese). Cantonese has no ISO-639-1 representation so we've included Cantonese variations for `Time` within the Chinese rules. But Cantonese has a ISO-639-3 representation (`yue`).

From a locale perspective, as Cantonese is an official language in Hongkong and Macau, it is represented as `zh-HK` and `zh-MO`.
Locales capture regional particularities of a language. For example, in French the numeral 70 is pronounced "soixante-dix" in France (`fr-FR`), and "septante" in Belgium (`fr-BE`) and Switzerland (`fr-CH`).

To summarize, it seems like we could add ISO-639-3 languages if they don't derive from another ISO-639-1 (macro)language (e.g. `fil` for Filipino), and support locales as language extensions (e.g. `zh-HK`/`zh-MO` for Cantonese, `fr-BE` for French in Belgium).
